From dfcd56cd503c0bc2f8e6f1a4672c4626ad312aba Mon Sep 17 00:00:00 2001
From: "Nagamani Chennuboina nagamani.chennuboina@intel.com"
 <nagamani.chennuboina@intel.com>
Date: Thu, 27 May 2021 14:03:50 +0530
Subject: [PATCH] Changed threading to oneTBB for InferenceEngine

Added support for oneTBB libs

Tracked-On: OAM-97456
Signed-off-by: Nagamani Chennuboina <nagamani.chennuboina@intel.com>
---
 Android.bp | 53 ++++++++++++++++++++++++++++++++++++++---------------
 1 file changed, 38 insertions(+), 15 deletions(-)

diff --git a/Android.bp b/Android.bp
index a5ee1c289644..dd2516869a72 100644
--- a/Android.bp
+++ b/Android.bp
@@ -134,7 +134,7 @@ cc_library_shared {
         "-DENABLE_OPENCV=OFF",
         "-DENABLE_MKL_DNN=1",
         "-DCI_BUILD_NUMBER=\"custom_HEAD_35384a63a6e4e2f2ac94277ba8f70a57276d0231\"",
-        "-DIE_THREAD=IE_THREAD_OMP",
+        "-DIE_THREAD=IE_THREAD_TBB",
         "-DGAPI_STANDALONE",
         "-DIE_BUILD_POSTFIX=\"\"",
         "-DUSE_STATIC_IE",
@@ -150,17 +150,19 @@ cc_library_shared {
         "liblog",
         "libngraph",
         "libinference_engine_transformations",
-        "libc++_shared"
+        "libc++_shared",
+        "libtbb",
+        "libtbbmalloc"
     ],
 
     static_libs: [
         "libitt",
         "libpugixml",
-        "libomp",
     ],
 
     header_libs: [
         "libmkldnn_headers",
+        "libtbb_headers",
     ],
     strip: {
         none: true,
@@ -303,7 +305,7 @@ cc_library_shared {
         "-DENABLE_GNA=1",
         "-DIE_BUILD_POSTFIX=\"\"",
         "-DNGRAPH_VERSION=\"0.0.0+a9c6e72\"",
-        "-DIE_THREAD=IE_THREAD_OMP",
+        "-DIE_THREAD=IE_THREAD_TBB",
         "-DGAPI_STANDALONE",
         "-DIMPLEMENT_INFERENCE_ENGINE_API",
         "-Dinference_engine_legacy_EXPORTS"
@@ -312,7 +314,6 @@ cc_library_shared {
 
     static_libs: [
         "libpugixml",
-        "libomp",
         "libitt",
     ],
 
@@ -320,7 +321,9 @@ cc_library_shared {
         "liblog",
         "libinference_engine_transformations",
         "libngraph",
-        "libinference_engine"
+        "libinference_engine",
+        "libtbb",
+        "libtbbmalloc"
     ],
 
     header_libs: [
@@ -329,6 +332,7 @@ cc_library_shared {
         "libinference_headers",
         "libngraph_headers",
         "libpugixml_headers",
+        "libtbb_headers",
     ],
     strip: {
         none: true,
@@ -430,7 +434,7 @@ cc_library_shared {
         "-DENABLE_MKL_DNN=1",
         "-DENABLE_GNA=1",
         "-DIE_BUILD_POSTFIX=\"\"",
-        "-DIE_THREAD=IE_THREAD_OMP",
+        "-DIE_THREAD=IE_THREAD_TBB",
         "-DIMPLEMENT_INFERENCE_ENGINE_API",
         "-Dinference_engine_lp_transformations_EXPORTS",
     ],
@@ -440,10 +444,16 @@ cc_library_shared {
         "libitt",
     ],
 
+    header_libs: [
+        "libtbb_headers",
+    ],
+
     shared_libs: [
         "liblog",
         "libinference_engine_transformations",
         "libngraph",
+        "libtbb",
+        "libtbbmalloc"
     ],
 
     strip: {
@@ -567,7 +577,7 @@ cc_library_shared {
         "-DIE_BUILD_POSTFIX=\"\"",
         "-DNGRAPH_VERSION=\"0.0.0+35384a6\"",
         "-Dinference_engine_transformations_EXPORTS",
-        "-DIE_THREAD=IE_THREAD_OMP",
+        "-DIE_THREAD=IE_THREAD_TBB",
         "-Werror",
         "-Wall",
         "-Wno-strict-overflow",
@@ -579,7 +589,16 @@ cc_library_shared {
     ],
     rtti: true,
 
-    shared_libs: ["libngraph",  "liblog"],
+    shared_libs: [
+        "libngraph",
+        "liblog",
+        "libtbb",
+        "libtbbmalloc"
+    ],
+
+    header_libs: [
+        "libtbb_headers",
+    ],
 
     static_libs: [
         "libitt",
@@ -837,14 +856,14 @@ cc_library_shared {
         "-DENABLE_MKL_DNN=1",
         "-DENABLE_SPEECH_DEMO",
         "-DIE_BUILD_POSTFIX=\"\"",
-        "-DIE_THREAD=IE_THREAD_SEQ",
         "-DIMPLEMENT_INFERENCE_ENGINE_PLUGIN",
         "-DMKLDNNPlugin_EXPORTS",
-        "-DMKLDNN_THR=MKLDNN_THR_SEQ",
+        "-DIE_THREAD=IE_THREAD_TBB",
+        "-DMKLDNN_ENABLE_CONCURRENT_EXEC",
+        "-DMKLDNN_THR=MKLDNN_THR_TBB",
         "-DCI_BUILD_NUMBER=\"custom_HEAD_35384a63a6e4e2f2ac94277ba8f70a57276d0231\"",
         "-fvisibility=hidden",
         "-fvisibility-inlines-hidden",
-        "-fopenmp",
         "-Wno-error",
         "-Wno-delete-non-abstract-non-virtual-dtor",
     ],
@@ -857,16 +876,18 @@ cc_library_shared {
         "libinference_engine_lp_transformations",
         "libinference_engine_transformations",
         "libngraph",
+        "libtbb",
+        "libtbbmalloc"
     ],
 
     static_libs: [
         "libitt",
         "libmkldnn",
-        "libomp",
     ],
 
     header_libs: [
         "libmkldnn_headers",
+        "libtbb_headers",
     ],
     strip: {
         none: true,
@@ -986,6 +1007,7 @@ cc_library_shared {
         "libinference_legacy_headers",
         "libinference_lp_transformations_headers",
         "libngraph_headers",
+        "libtbb_headers",
     ],
 
     cflags: [
@@ -1004,7 +1026,6 @@ cc_library_shared {
         "-Wno-non-virtual-dtor",
         "-Wno-missing-field-initializers",
         "-Wno-error",
-        "-fopenmp",
         "-frtti",
         "-fexceptions",
         "-std=c++11",
@@ -1016,7 +1037,7 @@ cc_library_shared {
         "-D__ANDROID__",
         "-DNNLOG",
         "-DDEBUG",
-        "-DIE_THREAD=IE_THREAD_OMP",
+        "-DIE_THREAD=IE_THREAD_TBB",
         "-DCI_BUILD_NUMBER=\"custom_HEAD_34ddb70f7d918e787b387870609f613b45f42dde\"",
         "-DIMPLEMENT_INFERENCE_ENGINE_PLUGIN",
         "-DGNA_LIB_VER=2",
@@ -1031,6 +1052,8 @@ cc_library_shared {
         "libinference_engine_lp_transformations",
         "libinference_engine_legacy",
         "libgna",
+        "libtbb",
+        "libtbbmalloc"
     ],
 
     srcs: [
-- 
2.31.1

